SimCalorimetry/HcalTrigPrimProducers/python/hcaltpdigi_cff.py

```diff
@@ -4,13 +4,17 @@
 
 import FWCore.ParameterSet.Config as cms
 
+from Configuration.Eras.Modifier_run2_HCAL_2018_cff import run2_HCAL_2018
+
 from SimCalorimetry.HcalTrigPrimProducers.hcaltpdigi_cfi import *
 from CalibCalorimetry.CaloTPG.CaloTPGTranscoder_cfi import *
 HcalTPGCoderULUT = cms.ESProducer("HcalTPGCoderULUT",
     read_Ascii_LUTs = cms.bool(False),
     read_XML_LUTs = cms.bool(False),
     read_FG_LUTs = cms.bool(False),
     LUTGenerationMode = cms.bool(True),
+    linearLUTs = cms.bool(False),
+    TPScales = TPScales,
```

Comment:
CMS naming convention would suggest names starting with lower case, I think.
This would have to be propagated to the configs, of course.

Reply:
Whenever I try to look for a consistent CMS style, I get lost… is there something that a mere mortal can find within 60s of using Google? The FAQ on GH seem to be more concerned with CVS than with coding style!

I can rename *that* parameter (and more, if requested)… at some point this is going to become a gigantic code-reformatting exercise, though.

Reply:
I am also not aware of anything else than this:
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideDevelopersGuide#Code_conventions
But as it says, the linked document is a bit outdated.
I actually make these comments on the convention when I saw such comments by others with a longer history in CMS code development, e.g. @slava77 makes a lot of such comments and might know a more up-to-date reference.

Reply:
I think I read through those a few years back, when Jim Brooke pointed them out to me, and they are no more than common sense to me, and kept pretty vague, still. IMO it is at least missing a clarification of using camel case vs underscores (in variables), and that variable names should not look like classes, i.e., should start with a lowercase letter. I fixed one such case in the last commit.

A major omission is the lack of a whitespace definition: indentation, parenthesis placement, whitespace around operators, no one-line if/switch/case statements… I parse code partially by its visual structure, and having a complete mix of indentation styles is pretty annoying. `mkedanlzr` spat out a file with 3 space indent when I used it first, and since no-one seemed to agree on anything, that's what I use as a standard for CMSSW. I'm amazed that no-one is bothered enough by the constant re-formatting to actually enforce something homogeneous!\</rant>

Reply:
there's a syntax error: you are missing the opening `<rant>` tag ;)

Reply:
Customary syntax error… :)

Anyhow, do you want me to change that `TPScales` to `tpScales`?

Reply:
FWIW, I would like to do so in a follow-up PR. I did a good axe-grind and removed a bunch of HCAL LUT-generation code that should never be able to run anyways, this would be a candidate for that, too.

Reply:
I would prefer that you change this name here as you are introducing the parameter here.
Other parameters where this applies could be handled in a separate PR.

Reply:
OK